lib/proto/http.js

```diff
@@ -27,7 +27,8 @@ module.exports = {
         let fallbackRetryDelayInMs = ms(opts.fallbackRetryDelay || '60s');
 
         const options = {
-            uri: encodeURI(link),
+            // Decoding and encoding is required to prevent encoding already encoded URLs
+            uri: encodeURI(decodeURI(link)),
```

Comment:
The encoding passes should be opt-in so that people can catch places were they forgot to encode their URLs rather than silently passing because of this

Reply:
It's not for people forgetting to encode but to support unicode.
Url supports unicode but on the server side, escaped unicode is required.
See #23